src/Inciter/FaceData.h

```diff
@@ -0,0 +1,124 @@
+// *****************************************************************************
+/*!
+  \file      src/Inciter/FaceData.h
+  \copyright 2012-2015, J. Bakosi, 2016-2018, Los Alamos National Security, LLC.
+  \details   Face-data used only in discontinuous Galerkin discretization scheme
+*/
+// *****************************************************************************
+#ifndef FaceData_h
+#define FaceData_h
+
+#include "PUPUtil.h"
+#include "UnsMesh.h"
+#include "Inciter/InputDeck/InputDeck.h"
+
+namespace tk {
+  class ExodusIIMeshWriter;
+  class RootMeshWriter;
+}
+
+namespace inciter {
+
+extern ctr::InputDeck g_inputdeck;
+
+//! \brief FaceData class holding face-connectivity data useful
+//!   to DG discretization
+class FaceData
+{
+  public:
+    //! Empty constructor for Charm++
+    explicit FaceData() {}
+
+    //! Constructor
+    explicit
+      FaceData(
+        const std::vector< std::size_t >& conn,
+        std::size_t nbfac_complete,
+        const std::map< int, std::vector< std::size_t > >& bface,
+        const std::vector< std::size_t >& triinpoel_complete );
+
+    /** @name Accessors
+      * */
+    ///@{
+    const std::vector< std::size_t >& Gid() const { return m_gid; }
+    std::vector< std::size_t >& Gid() { return m_gid; }
+
+    const std::unordered_map< std::size_t, std::size_t >& Lid() const
+    { return m_lid; }
+    std::unordered_map< std::size_t, std::size_t >& Lid() { return m_lid; }
+
+    const std::vector< std::size_t >& Inpoel() const { return m_inpoel; }
+    std::vector< std::size_t >& Inpoel() { return m_inpoel; }
```

Comment:
Remember that `m_el` (and thus all of `m_inpoel`, `m_gid`, and `m_lid`) is a duplicate here. Accordingly, let's not even store these in `FaceData`: create it as a local variable in the constructor, use them there and it will be correctly thrown away at the end of the scope. Thus these accessors, `Gid`, `Lid`, and `Inpoel`, should also go away.